front/src/components/ProductCatalogue/ProductRating.js

```diff
@@ -1,41 +1,91 @@
 import React, { useEffect, useState } from "react";
-import { Box, Rating } from "@material-ui/core";
+import { Box } from "@material-ui/core";
+import axios from "axios";
+import Rating from "@material-ui/lab/Rating";
+import { useSelector } from "react-redux";
 
 function ProductRating(props) {
-  // const { user } = useSelector(state => state.account);
-  const [rating, setRating] = useState(0);
+  const { user } = useSelector((state) => state.account);
+  const [avgRating, setAvgRating] = useState(2);
+  const [count, setCount] = useState(1);
+  const [isRated, setIsRated] = useState(false); //flag for user input
+  const [isReadOnly, setIsReadOnly] = useState(false);
+
+  const handleIsReadOnly = (value) => {
+    if (!user) {
+      setIsReadOnly(true);
+      console.log("osiujdfoisdf");
+    } else {
+      setIsReadOnly(value);
+    }
+  };
+
+  console.log(!user, user);
 
   useEffect(() => {
-    // cia reikia call'o i back'a pasiimti produkto ratinga;
-    const getRating = async () => {
-      // call get rating + props.id;
-      // await call;
-      // setRating(response);
-    };
-    getRating();
-  }, []);
-
-  //   const async handleRatingChange = (value) => {
-  //      funkcija, kuri handlina zmogaus reitingavima;
-  //      paimam value ir kazka dar, nezinau, reik paziuret back'o swaggeri.
-  //      mes turbut darom post'a i back'a kad pareitinguot produkta.
-  //      const data = {
-  //         rating: value
-  //      }
-  //      const response = await axios.post("URL", data, {headers: pakopink is manes} );
-  //      console.log(response);
-  //   }
+    handleIsReadOnly(false); //initial readonly check
+
+    async function getProductRating() {
+      try {
+        const response = await axios.get(
+          "/review/71a4e7f6-333b-4dd1-997f-ce38efa3aee4"
+        );
+        // const rating = isRated ? 1.5 : response.data.averageScore;
+        // setAvgRating(rating);
+        setAvgRating(response.data.averageScore);
+        setCount(response.data.count);
+      } catch (error) {}
+    }
+
+    async function getUserRating() {
+      try {
+        const response = await axios.get("/review", {
+          params: {
+            productId: "71a4e7f6-333b-4dd1-997f-ce38efa3aee4",
+          },
+        });
+        if (response.status === 200) {
+          handleIsReadOnly(true);
+        }
+      } catch (error) {
+        if (error.response.status === 404) {
+          handleIsReadOnly(false);
+        }
+      }
+    }
+
+    getProductRating();
+    getUserRating();
+  }, [isRated, user]);
```

Comment:
should check if it isin't looping forever (endless cycle). try console.log("hit") and check the console output.

Reply:
console.log should be inside useEffect line 56.